ndnfs/driver/driver.go

```diff
@@ -0,0 +1,360 @@
+		} else {
+			service = d.Config.Servicename
+		}
+	} else {
+		service = r.Options["service"]
+	}
+	data["bucketName"] = r.Name
+	url := fmt.Sprintf("clusters/%s/tenants/%s/buckets", cluster, tenant)
+
+	body, err := d.Request("POST", url, data)
+	resp := make(map[string]interface{})
+	jsonerr := json.Unmarshal(body, &resp)
+	if (jsonerr != nil) {
+		log.Error(jsonerr)
+	}
+	if (resp["code"] != nil) && (resp["code"] != "RT_ERR_EXISTS") {
+		err = errors.New(fmt.Sprintf("Error while handling request: %s", resp))
+		return volume.Response{Err: err.Error()}
+	}
+
+	data = make(map[string]interface{})
+	data["chunkSize"] = chunkSizeInt
+	data["serve"] = filepath.Join(cluster, tenant, r.Name)
+	url = fmt.Sprintf("service/%s/serve", service)
+	body, err = d.Request("PUT", url, data)
+	resp = make(map[string]interface{})
+	jsonerr = json.Unmarshal(body, &resp)
+	if (jsonerr != nil) {
+		log.Error(jsonerr)
+	}
+	if resp["code"] == "EINVAL" {
+		err = errors.New(fmt.Sprintf("Error while handling request: %s", resp))
+		return volume.Response{Err: err.Error()}
+	}
+
+	return volume.Response{}
+}
+
+func (d NdnfsDriver) Get(r volume.Request) volume.Response {
+	log.Debug(DN, "Get volume: ", r.Name, " Options: ", r.Options)
+	var mnt string
+	nfsMap, err := d.ListVolumes()
+	if err != nil || r.Name == "" {
+		log.Info("Volume with name ", r.Name, " not found")
+		return volume.Response{}
+	}
+
+	for k, v := range nfsMap {
+		if k == r.Name {
+			mnt = v
+			break
+		}
+	}
+	if mnt == "" {
+		return volume.Response{}
+	}
+
+	log.Debug("Device mountpoint is: ", mnt)
+	return volume.Response{Volume: &volume.Volume{
+		Name: r.Name, Mountpoint: mnt}}
+}
+
+func (d NdnfsDriver) List(r volume.Request) volume.Response {
+	log.Debug(DN, "List volume")
+	vmap, err := d.ListVolumes()
+	if err != nil {
+		log.Info("Failed to retrieve volume list", err)
+		return volume.Response{Err: err.Error()}
+	}
+	log.Debug(DN, "Nedge response: ", vmap)
+	var vols []*volume.Volume
+	for name, mnt := range vmap {
+		if name != "" {
+			vols = append(vols, &volume.Volume{Name: name, Mountpoint: mnt})
+		}
+	}
+	return volume.Response{Volumes: vols}
+}
+
+func (d NdnfsDriver) Mount(r volume.MountRequest) volume.Response {
+	log.Info(DN, "Mount volume: ", r.Name)
+	d.Mutex.Lock()
+	defer d.Mutex.Unlock()
+	var err error
+	var mnt string
+
+	nfs := fmt.Sprintf("%s:/%s", d.Config.Nedgedata, r.Name)
+	mnt = filepath.Join(d.Config.Mountpoint, r.Name)
+	log.Debug(DN, "Creating mountpoint folder: ", mnt)
+	if out, err := exec.Command("mkdir", "-p", mnt).CombinedOutput(); err != nil {
+	    log.Info("Error running mkdir command: ", err, "{", string(out), "}")
+	}
+	log.Debug(DN, "Mounting Volume ", r.Name)
+	args := []string{"-t", "nfs", nfs, mnt}
+	if out, err := exec.Command("mount", args...).CombinedOutput(); err != nil {
+		log.Error("Error running mount command: ", err, "{", string(out), "}")
+		err = errors.New(fmt.Sprintf("%s: %s", err, out))
+		if err != nil {
```

Comment:
Looks suspicious. Maybe checking err for nil should be before creating a new err from the old err's content